src/client/Microsoft.Identity.Client/Platforms/Features/DesktopOS/WindowsNativeUtils.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Identity.Client.Platforms.Features.DesktopOs
+{
+    /// <summary>
+    /// Public Windows native methods
+    /// </summary>
+    public static class WindowsNativeUtils
+    {
+        /// <summary>
+        /// Tests whether the current user is a member of the Administrator's group.
+        /// </summary>
+        /// <returns>True if the current user is an Admin; false, otherwise.</returns>
+        public static bool IsElevatedUser() => IsUserAnAdmin();
+
+        /// <summary>
+        /// Registers security and sets the security values for the process.
+        /// </summary>
+        /// <remarks>
+        /// Workaround to enable WAM Account Picker in an elevated process.
+        /// </remarks>
+        public static void InitializeProcessSecurity()
+        {
+            int result = CoInitializeSecurity(
+                IntPtr.Zero, -1, IntPtr.Zero,
+                IntPtr.Zero, RpcAuthnLevel.None,
+                RpcImpLevel.Impersonate, IntPtr.Zero,
+                EoAuthnCap.None, IntPtr.Zero);
+
+            if (result != 0)
+            {
+                throw new MsalClientException(MsalError.InitializeProcessSecurityError, MsalErrorMessage.InitializeProcessSecurityError($"0x{result:x}"));
+            }
+        }
+
+        [DllImport("shell32.dll")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        private static extern bool IsUserAnAdmin();
```

Comment:
Another way to do this is `new WindowsPrincipal(WindowsIdentity.GetCurrent()).IsInRole(WindowsBuiltInRole.Administrator)`; however, it's [not available](https://apisof.net/catalog/7716df7f-41fa-ebc8-1c5a-bd4d110b7ed1) in netcoreapp2.1.